csrank/objectranking/list_net.py

```diff
@@ -90,9 +84,6 @@ def __init__(
                 [1] Z. Cao, T. Qin, T. Liu, M. Tsai and H. Li. "Learning to Rank: From Pairwise Approach to Listwise Approach." ICML, 2007.
         """
         self.logger = logging.getLogger(ListNet.__name__)
-        self.n_object_features = n_object_features
-        self.n_objects = n_top
-        self.n_top = n_top
```

Comment:
This may need review. I have assumed `n_top` is essentially equivalent to what is called `n_objects` in other estimators and can be inferred from the data as well. Is that correct? Why does it have a different name?

Reply:
ListNet as defined in the original paper only calculates the top-1 ranking probability in order to define a suitable and efficient loss function. This later was generalized to top-k rankings.
The `n_top` here is the k and specifies how many objects to consider for the loss function.
This is why setting `n_top = n_objects` is not the same thing.

Reply:
Ah, okay. Its a bit confusing that the original code set `n_objects` to `n_top`, but then `n_objects` wasn't used anywhere either.

Since we still need to avoid mandatory init arguments, how should we handle this case? Would it be reasonable to default to `None` and then fill that in with `n_objects` on `fit`? Or, going with the initial formulation in the paper, default to `1` instead?

Reply:
I would be in favor of a default value. `1` sounds fine.

Reply:
Okay, thanks for the feedback. I have implemented the default value of `1`, please have another look.